code/modules/food_and_drinks/drinks/drinks/bottle.dm

```diff
@@ -537,3 +537,44 @@
 		to_chat(user, "<span class='info'>You snuff out the flame on [src].</span>")
 		cut_overlay(GLOB.fire_overlay)
 		active = 0
+
+/obj/item/reagent_containers/food/drinks/bottle/pruno
+	name = "pruno mix"
+	desc = "A trash bag filled with fruit, sugar, yeast, and water, pulped together into a pungent slurry to be fermented in an enclosed space, traditionally the toilet. Security would love to confiscate this, one of the many things wrong with them."
+	icon = 'icons/obj/janitor.dmi'
+	icon_state = "trashbag"
+	list_reagents = list(/datum/reagent/consumable/prunomix = 50)
+	var/fermentation_time = 30 SECONDS // time between fermentation checks, see below
+
+/obj/item/reagent_containers/food/drinks/bottle/pruno/Initialize()
+	. = ..()
+	addtimer(CALLBACK(src, .proc/check_fermentation), fermentation_time)
+
+// Checks to see if the pruno can ferment, i.e. is it inside a structure (e.g. toilet), or a machine (e.g. washer)?
+// if so, start countdown to ferment, if not, check again in fermentation_time seconds
+// TODO: make it so the washer spills reagents if a reagent container is in there, for now, you can wash pruno
+
+/obj/item/reagent_containers/food/drinks/bottle/pruno/proc/check_fermentation()
+	if (!(istype(loc, /obj/machinery) || istype(loc, /obj/structure)))
```

Comment:
you should be using movement signals to start/cancel fermentation, not checking every x seconds